distributed/protocol/serialize.py

```diff
@@ -32,20 +32,22 @@ def dask_dumps(x, context=None):
     except TypeError:
         raise NotImplementedError(type_name)
     if has_keyword(dumps, "context"):
-        header, frames = dumps(x, context=context)
+        sub_header, frames = dumps(x, context=context)
     else:
-        header, frames = dumps(x)
+        sub_header, frames = dumps(x)
 
+    header = {}
+    header["sub-header"] = sub_header
     header["type"] = type_name
-    header["type-serialized"] = pickle.dumps(type(x), protocol=4)
+    header["type-serialized"] = pickle.dumps(type(x))
```

Comment:
I think this was needed to support downstream projects that read `"type-serialized"` but didn't support `protocol=5`

Reply:
Since `distributed` still supports Python 3.7, shouldn't we keep `protocol=4` then?

Reply:
> Since `distributed` still supports Python 3.7, shouldn't we keep `protocol=4` then?
Now that we don't overwrite `"type-serialized"`, downstream projects that writes and reads `"type-serialized"` using the same pickle version.

Reply:
cc @mrocklin (in case you have thoughts here as I think this was something you added)